torch/_dynamo/config.py

```diff
@@ -118,6 +118,10 @@
 # Show a warning for every specialization
 print_specializations = False
 
+# Simplify guards, summarizing static and dynamic constraints on dimensions.
+# NOTE: This only has an effect when dynamic_shapes=True.
+summarize_dim_constraints = False
```

Comment:
Why is this gated, btw? If it's generally good and able, we should include it unequivocally, right? Or what is the tradeoff space for enabling this?

Reply:
I'd like to leave this gated for export only, at least until we have been using it for a while. This is mainly because I have only tested this in the wild on export use cases, and I don't know (a) what performance bar we need to meet for eager (b) what corner cases we could be missing that could blow up these solvers, which we haven't used before.